### PR TITLE
Navbar UI

### LINKED ISSUE #17: Create dashboard UI
## Description
As a user, I need to see the dashboard once I sucessfully login to the website.

## Mocks
As finalized in #5 

### Update [Required]
- [ ] There is a main navbar with settings and logout options
- [ ] The user's personal information is displayed
- [ ] There is an inner navbar with the 3 tabs: Dashboard, forms, submissions.

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 2 mentors.

## Estimation
2 days

### PR DESCRIPTION
**Pull request by bismitaguha**
### Description
- Added Settings and Logout (Settings is a dummy button)

Fixes #17 #5 
### Type of Change:

- Code
- User Interface

**Code/Quality Assurance Only**
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?

Screenshot from 2020-06-18 19-42-32

The User dashboard
Screenshot from 2020-06-20 22-23-27

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [ ] Any dependent changes have been merged

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings
- [ ] Any dependent changes have been published in downstream modules

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -42,5 +42,8 @@
       "last 1 firefox version",
       "last 1 safari version"
     ]
+  },
+  "devDependencies": {
```
Comment (abha224) [inline package.json]: @bismitaguha I think @sidvenu asked you to shift this to another pr. He has created an issue also for the same. Pls check

File: src/AuthRoute.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+export const AuthRoute = ({ component: Component, ...rest }) => (
+    <>
+    <Navbar/>
+    <Route {...rest} render={props => (
+        localStorage.getItem('token') === null
+        ? <Component {...props} />
+        : <Redirect to='/'/>
+    )} />
+    </>
+)
```
Comment (abha224) [inline src/AuthRoute.js]: newline missing

File: src/actions/info.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            payload: res.data
+        });
+    }
+    catch (err) {
+        dispatch({
+            type: USER_INFO_ERRORS,
+            payload: err.response.data
+        });
+    }
+};
```
Comment (abha224) [inline src/actions/info.js]: newline missing

File: src/actions/login.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        callback()
+    }
+    catch (err) {
+        dispatch({
+            type: REGISTER_ERRORS,
+            payload: err.response.data
+        });
+        callback()
+    }
+};
```
Comment (abha224) [inline src/actions/login.js]: newline missing

File: src/components/App.js [hunk 5]
```diff
@@ -1,19 +1,28 @@
...
+import React, { Component } from 'react'
+// import { connect } from 'react-redux'
+import { Route, Switch, Router } from 'react-router-dom'
+import Routes from '../Routes'
+// import Navbar from './Navbar'
+import { history } from '../store'
 import app from '../styles/App.css'
 
+
+// const store = configureStore()
```
Comment (abha224) [inline src/components/App.js]: Pls remove the comments in this file

File: src/components/Login.js [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+        this.submitLogin = this.submitLogin.bind(this);
+        this.onChange = this.onChange.bind(this);
+    }
+
+    handleShow = (e) => this.setState({
+        showPassword: !this.state.showPassword
+    })
+
+    submitLogin = () => {
+        console.log("hello")
```
Comment (abha224) [inline src/components/Login.js]: pls remove the debug statements

File: src/components/Form.js [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+
+export default class Form extends Component {
+    render() {
+        return (
+            <div>
+                forms
+            </div>
+        )
+    }
+}
```
Comment (abha224) [inline src/components/Form.js]: newline

File: src/components/Login.js [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+
+const mapStateToProps = state => ({
+    login: state.login.login,
+    loginerror: state.login.loginerror
+})
+
+export default connect(
+    mapStateToProps,
+    { postLogin, }
+)(Login)
```
Comment (abha224) [inline src/components/Login.js]: newline missing

File: src/components/Menubar.js [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+};
+
+const mapStateToProps = state => ({
+    userinfo: state.info.userinfo,
+})
+
+export default connect(
+    mapStateToProps,
+    { getInfo, }
+)(Menubar)
```
Comment (abha224) [inline src/components/Menubar.js]: newline missing

File: src/components/Navbar.js [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+    userinfo: state.info.userinfo,
+    userinfoerror: state.info.userinfoerror,
+    user: state.user.user,
+    usererror: state.user.usererror,
+})
+
+export default connect(
+    mapStateToProps,
+    { getInfo, getUser, }
+)(Navbar)
```
Comment (abha224) [inline src/components/Navbar.js]: newline missing

File: src/components/Register.js [hunk 11]
```diff
@@ -0,0 +1,265 @@
...
+
+const mapStateToProps = state => ({
+    register: state.login.register,
+    registererror: state.login.registererror
+})
+
+export default connect(
+    mapStateToProps,
+    { postRegister, }
+)(Register)
```
Comment (abha224) [inline src/components/Register.js]: newline missing

File: src/components/Submission.js [hunk 12]
```diff
@@ -0,0 +1,11 @@
...
+
+export default class Form extends Component {
+    render() {
+        return (
+            <div>
+                submissions
+            </div>
+        )
+    }
+}
```
Comment (abha224) [inline src/components/Submission.js]: newline missing

File: src/reducers/index.js [hunk 13]
```diff
@@ -0,0 +1,12 @@
...
+import infoReducer from './info'
+import userReducer from './user'
+
+const rootReducers = combineReducers ({
+    login: loginReducer,
+    info: infoReducer,
+    user: userReducer
+})
+
+export default rootReducers
```
Comment (abha224) [inline src/reducers/index.js]: newline missing

File: src/styles/Dashboard.css [hunk 14]
```diff
@@ -0,0 +1,18 @@
...
+.left {
+    display: flex;
+    flex-direction: column;
+    width: 70vw;
+}
+
+.right {
+    display: flex;
+    flex-direction: column-reverse;
+    /* width: 10vw; */
```
Comment (abha224) [inline src/styles/Dashboard.css]: remove this comment

**Review by abha224 — COMMENTED** [review]
@bismitaguha Pls add newlines wherever necessary, a lot of the files still dont have them. I have not commented on all of the file, so check all of them once. Also, remove the code that you have commented out which are not necessary for the pr.

**Comment by bismitaguha** [comment]
@abha224 Changes made as suggested! :+1:

**Review by abha224 — APPROVED** [review]
LGTM :+1:

File: src/PrivateRoute.js [hunk 15]
```diff
@@ -0,0 +1,19 @@
...
+export const PrivateRoute = ({ component: Component, ...rest }) => (
+    <>
+    <Navbar/>
+    <Route {...rest} render={props => (
+        localStorage.getItem('token')
+        ?
+        <>
+        <Menubar/>
+        <Component {...props} />
+        </>
```
Comment (sidvenu) [inline src/PrivateRoute.js]: Can you place this inside `()`? At the first look, the use of the ternary operator is confusing.

File: src/index.js [hunk 16]
```diff
@@ -15,48 +13,3 @@ ReactDOM.render (
...
-//       </Provider>
-//     )
-//   }
-// }
-
-
-
-// If you want your app to work offline and load faster, you can change
-// unregister() to register() below. Note this comes with some pitfalls.
-// Learn more about service workers: https://bit.ly/CRA-PWA
```
Comment (sidvenu) [inline src/index.js]: I don't think this is relevant to this PR. Please check.

**Review by sidvenu — CHANGES_REQUESTED** [review]

File: src/PrivateRoute.js (hunk 15 above)
Comment (bismitaguha) [reply src/PrivateRoute.js]: Not clear?? @sidvenu

Comment (sidvenu) [reply src/PrivateRoute.js]:
```suggestion
        (
          <>
          <Menubar/>
          <Component {...props} />
          </>
        )
```

**Comment by bismitaguha** [comment]
@sidvenu Suggestions resolved!

**Review by sidvenu — APPROVED** [review]

**Review by sidvenu — APPROVED** [review]

**Pull request merged**